### ISSUE TITLE
Keras Unable to clone_model a load_model result saved in TF format

### ISSUE THREAD

**Issue by dvbuntu** [tensorflow/tensorflow#37723]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): Yes, see attached example
- OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04): CentOS 7
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if
the issue happens on mobile device: N/A
- TensorFlow installed from (source or
binary): Binary
- TensorFlow version (use command below): 2.0.1
- Python version: 3.6.8
- Bazel
version (if compiling from source): N/A
- GCC/Compiler version (if compiling from
source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

Attempting to clone a loaded model that had been saved in TF format (i.e. a directory rather than `.h5`) creates an error.  This appears to be because the loaded model is of a slightly malformed type.

**Describe the expected behavior**

`clone_model` should be able to clone derived types.  `load_model` should return a true model instance even when loading a TF format model.

**Standalone code to reproduce the issue** 
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

```python
import tensorflow as tf

# make dummy functional model
x = tf.keras.layers.Input(10)
y = tf.keras.layers.Dense(10)(x)

a = tf.keras.models.Model(x,[y])

# save model with tf format
a.save('foo_model')

print('Type of a is {0}'.format(type(a)))
print('a._is_graph_network = {0}'.format(a._is_graph_network))


# read in the model
b = tf.keras.models.load_model('foo_model')

print('Type of b is {0}'.format(type(b)))
print('b._is_graph_network = {0}'.format(b._is_graph_network))

# Try to clone b, errors because b is not of correct type.
c = tf.keras.models.clone_model(b)
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

Output
```
Type of a is <class 'tensorflow.python.keras.engine.training.Model'>
a._is_graph_network = True
Type of b is <class 'tensorflow.python.keras.saving.saved_model.load.Model'>
b._is_graph_network = False
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/proj/keras_load_clone.py in <module>()
     21 
     22 # Try to clone b, errors because b is not of correct type.
---> 23 c = tf.keras.models.clone_model(b)

~/.local/lib/python3.6/site-packages/tensorflow_core/python/keras/models.py in clone_model(model, input_tensors, clone_function)
    420   else:
    421     return _clone_functional_model(
--> 422         model, input_tensors=input_tensors, layer_fn=clone_function)
    423 
    424 

~/.local/lib/python3.6/site-packages/tensorflow_core/python/keras/models.py in _clone_functional_model(model, input_tensors, layer_fn)
    163                      'got a `Sequential` instance instead:', model)
    164   if not model._is_graph_network:
--> 165     raise ValueError('Expected `model` argument '
    166                      'to be a functional `Model` instance, '
    167                      'but got a subclass model instead.')

ValueError: Expected `model` argument to be a functional `Model` instance, but got a subclass model instead.
```

I've tried overwriting the `_is_graph_network` attribute, but that just pushes off the problem to another line.

The reason I have to load and then clone is so I can recreate old models with specific datatypes (because TF 2.0.1 can't save mixed_float16 models, I had to save as float32 and fix after loading).  Normally, one would just rebuild the model from the actual commands, but I've made backward-compatibility breaking changes since then.  I'd like to still be able to use my old models.  I'm looking at just updating to TF2.2, but I don't control my environment so that may not be an option.

**Comment by amahendrakar** [comment]
@dvbuntu,
I was able to reproduce the issue with [TF v2.0.1](https://colab.research.google.com/gist/amahendrakar/f91d8c5a8e92207e3bc4fc44d69d321c/37723.ipynb) and [TF v2.1.0](https://colab.research.google.com/gist/amahendrakar/aded3fa39bcd45c04682257f92fe0023/37723-2-1.ipynb). However, I was able to run the code without any issues with [TF-nightly](https://colab.research.google.com/gist/amahendrakar/b25afc6fff1c83162b03bfcb2bada4fb/37723-tf-nightly.ipynb). Please find the attached gist. Thanks!

**Comment by dvbuntu** [comment]
Good to hear it's fixed in the 2.2.  I'll just this example to push for the upgrade.

**Comment by amahendrakar** [comment]
@dvbuntu,
Please feel free to close the issue if resolved. Thanks!
